src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
-package core.mate.academy.service;
-
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
-}
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Your implementation of MachineService.
+ */
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> bulldozers = new BulldozerProducer().get();
+            return new ArrayList<>(bulldozers);
+        }
+        if (type == Excavator.class) {
+            List<? extends Machine> excavators = new ExcavatorProducer().get();

```

Comment:
make producers - class fields. To avoid creating a new instance each time this method is called